File: leetcode75/python/Leet1207.py
#1207. Unique Number of Occurrence
#https://leetcode.com/problems/unique-number-of-occurrences/

class Solution:
    def uniqueOccurrences(self, arr: list[int]) -> bool:
        frequency_map = {}
        for num in arr:
            frequency_map[num] = frequency_map.get(num, 0) + 1

        frequency_set = set(frequency_map.values())

        return len(frequency_map) == len(frequency_set)


if __name__ == '__main__':
    arr = [1,2,2,1,1,3]

    solution = Solution()
    print(solution.uniqueOccurrences(arr))

File: leetcode75/java/Leet1207.java
package leetcode75.java;

import java.util.Map;
import java.util.Set;
import java.util.HashMap;
import java.util.HashSet;

//1207. Unique Number of Occurrence
//https://leetcode.com/problems/unique-number-of-occurrences/

public class Leet1207 {

    public boolean uniqueOccurrences(int[] arr) {

        Map<Integer, Integer> frequencyMap = new HashMap<>();
        for (int num : arr) {
            frequencyMap.put(num, frequencyMap.getOrDefault(num, 0) + 1);
        }

        Set<Integer> set = new HashSet<>();
        for (int num : frequencyMap.values()) {
            if (!set.add(num)) {
                return false;
            }
        }
        return true;

    }

    public static void main(String[] args) {
        int[] arr = { 1, 2, 2, 1, 1, 3 };

        Leet1207 leet1207 = new Leet1207();

        System.out.println(leet1207.uniqueOccurrences(arr));
    }

}